### PR TITLE
Reworks Viper Venom and regular Venom

### PR DESCRIPTION
**Pull request by colossusqw**
[LABEL][rework][ai][chemistry]

## About the PR 

- This PR renames Viper Venom to hemotoxin, and regular Venom to cytotoxin, as well as alter their behaviors in a few ways.
- Cytotoxin no longer has a flat gib chance, instead, it has a increasing counter to progressively remove limbs on the affected player, hemotoxin now has a recipe and lasts a bit longer on the player.
- Rattlesnakes have a lower bite cooldown, and unit per unit the hemotoxin is a bit more dangerous, but rattlesnakes now do dry warning bites if they haven't been bothered in a bit, instead of immediately going for injecting bites.
- Hemotoxin now has a recipe that does not produce a lot, and requires attention not to blow up.

## Why's this needed? 

- Venom and Viper Venom are a bit oddly specific for how generalized they are, and it would make sense to bring them on the same naming convention as neurotoxin.
- As for the changes in behavior, rattlesnakes being an insta bite if you haven't seen em shouldn't happen nearly as much, nor should them appearing ontop of people anymore, which justifies the slight bite beef up, for the cytotoxin, i wanted to give the venom something still very dangerous but not "random chance you're dead" level dangerous, and consistently making your limbs fall off achieves that while being just as horrifying.
- For the proposed recipe? It's a slow production that gets very close to ether's combustion temp, requiring the chemist to be very careful during the whole thing, and hemotoxin is one of the more fun off-meta options even if it doesn't stack well with other poisons, shouldn't be a problem balance wise.

## Changelog
```changelog
(u)Colossusqw
(+)Viper venom is now called Hemotoxin and has a recipe, 1 heparin, 3 ether, 2 hydrochloric acid at 75C
(+) Venom is now called cytotoxin, and has had it's gib chance swapped by progressively removing limbs from the affected mob.
(+)Rattlesnakes now have a slightly lower bite cooldown, but do warning strikes before actually biting.
```

### CODE REVIEW

File: code/modules/chemistry/Reagents-PoisonEtc.dm [hunk 1]
```diff
@@ -1355,37 +1357,59 @@ datum
 				if (prob(25))
 					M.reagents.add_reagent("histamine", rand(5,10) * mult)
 				if (our_amt < 20)
-					M.take_toxin_damage(1 * mult)
-					random_brute_damage(M, 1 * mult)
+					M.take_toxin_damage(0.75 * mult)
+					delimb_counter += 0.5 * mult
```
Comment (Tarmunora) [inline code/modules/chemistry/Reagents-PoisonEtc.dm:1361]:
```suggestion
```
>6u of venom would cause a guaranteed delimb with this unless purged.

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/modules/chemistry/Reagents-PoisonEtc.dm (hunk 1 above)
Comment (colossusqw) [reply code/modules/chemistry/Reagents-PoisonEtc.dm:1361]: changed to a +0.2, you need at least 15 units and 2:30+ minutes for a single limb loss then, if it's still too much, i can just remove it as requested

File: code/modules/chemistry/Reagents-PoisonEtc.dm [hunk 2]
```diff
@@ -1355,37 +1357,59 @@ datum
...
 
-				if (our_amt > 40 && probmult(4))
-					M.visible_message(SPAN_ALERT("<B>[M]</B> starts convulsing violently!"), "You feel as if your body is tearing itself apart!")
-					M.setStatusMin("knockdown", 15 SECONDS * mult)
+				if (delimb_counter > 15)
+					delimb_counter = 0
+
+					M.visible_message(SPAN_ALERT("<B>[M]</B> seems to be melting away!"), "You feel as if your body is tearing itself apart!")
+					M.setStatusMin("knockdown", 4 SECONDS * mult)
 					M.make_jittery(1000)
```
[the remark this answers is not in the document]
Comment (colossusqw) [reply code/modules/chemistry/Reagents-PoisonEtc.dm:1383]: reduced to 400

File: code/modules/chemistry/Reagents-PoisonEtc.dm [hunk 3]
```diff
@@ -1355,37 +1357,59 @@ datum
...
+		harmful/hemotoxin
+			name = "hemotoxin"
+			id = "hemotoxin"
 			description = "A dangerous toxin that causes massive bleeding and tissue damage"
 			reagent_state = LIQUID
 			fluid_r = 210
 			fluid_g = 180
 			fluid_b = 25
-			depletion_rate = 0.3
+			depletion_rate = 0.2
```
[the remark this answers is not in the document]
Comment (colossusqw) [reply code/modules/chemistry/Reagents-PoisonEtc.dm:1412]: gave my reasoning on the discord, but for a not so dangerous chem that is mostly flavor/can't be stacked with other chems barred heparin, a rate like this both gives it some reason for people to want to make it, and offsets the rattlesnake bites' initial dry warning bite. if you feel it should still be a 0.3 though, i can change it back

File: code/modules/chemistry/Reagents-PoisonEtc.dm (hunk 1 above)
Comment (colossusqw) [reply code/modules/chemistry/Reagents-PoisonEtc.dm:1361]: actually, yeah i don't see why it even needs to count towards that in low doses anyway, away that one goes

File: code/modules/chemistry/Reagents-PoisonEtc.dm [hunk 4]
```diff
@@ -1355,37 +1357,59 @@ datum
...
-							M.gib()
+					if (!isdead(M))
+						M.emote(pick("cry", "tremble", "scream"))
+
+					if(ishuman(M))
+						var/mob/living/carbon/human/H = M
+						take_bleeding_damage(H, null, rand(15,35) * mult, DAMAGE_STAB)
+
+						for (var/chosen_limb in limb_list)
+							var/obj/item/parts/limb = H.limbs.vars[chosen_limb]
```
[the remark this answers is not in the document]
Comment (colossusqw) [reply code/modules/chemistry/Reagents-PoisonEtc.dm]: added

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**